### PR TITLE
Change CODEOWNERS to PalmTree / C&I teams

### PR DESCRIPTION
**Pull request by izgeri**
### What does this PR do?
Updates CODEOWNERS from @cyberark/conjur-core-team to @cyberark/community-and-integrations-team and @cyberark/palmtree, since the latter two groups are in practice the primary maintainers of this content.

### What ticket does this PR close?
n/a

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

#### Manual tests
**If you are preparing for a release**, have you run the following manual tests to verify existing functionality continues to function as expected?
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local authn-k8s client image build

### CODE REVIEW

**Review by Tovli — APPROVED** [review]
LGTM

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,4 +1,5 @@
-* @cyberark/conjur-core-team @conjurinc/conjur-core-team @conjurdemos/conjur-core-team
+* @cyberark/community-and-integrations-team
+* @cyberark/palmtree
```
Comment (sgnn7) [inline .github/CODEOWNERS:2]: Format I think requires them to be on the same line: https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners#example-of-a-codeowners-file

**Review by sgnn7 — CHANGES_REQUESTED** [review]
@izgeri One small comment. Is there a reason that the other files need the owner repeated 3 times?

File: .github/CODEOWNERS (hunk 1 above)
Comment (izgeri) [inline .github/CODEOWNERS:2]:
```suggestion
* @cyberark/community-and-integrations-team @cyberark/palmtree
```

Comment (izgeri) [reply .github/CODEOWNERS]: fixed - which means this would be good to have a squash merge on

Comment (sgnn7) [reply .github/CODEOWNERS]: can we trim the other lines to have just 1 repetition of the reviewer team?

**Review by sgnn7 — APPROVED** [review]

**Pull request merged**